app/components/CurrencyAmount.tsx

```diff
@@ -155,18 +156,21 @@ export default function CurrencyAmount({
       >
         <Flex direction="row" alignContent="center" minWidth={displayMinWidth}>
           {noImage ? <></> : currencyIcon(currency, iconHeight)}
-          {/* @ts-ignore */}
-          <StatNumber
-            color={displayNumberColor}
-            fontSize={amountFontSize}
-            overflow="hidden"
-            // @ts-ignore
-            textOverflow="ellipsis"
-            whiteSpace="nowrap"
-          >
-            {displayAmount}
-            {showCurrencyText ? ` ${currency}` : <></>}
-          </StatNumber>
+          {displayAmount === '0' || displayAmount.startsWith('-') ? (
```

Comment:
Might want to return a "loading" state from the hook and make this conditional on that?

Reply:
In here we don't know that hook, the parent is responsible for that. The `CurrencyAmount` is the component that takes care of rendering all the currency values. Not sure if this design is optimal.

Reply:
What balance get displayed if you receive 0 or (-)? The previous one?

do note that "0" is a valid balance. How do you make a difference between actual 0 and error 0?

Reply:
Very good point! I took a step back and am implementing it more properly.